lib/dep-graph-builders/yarn-lock-v1/build-depgraph-simple.ts

```diff
@@ -73,18 +76,51 @@ const dfsVisit = async (
       includeOptionalDeps,
     );
 
-    if (!visitedMap.has(childNode.id)) {
-      addPkgNodeToGraph(depGraphBuilder, childNode, {});
-      await dfsVisit(
-        depGraphBuilder,
-        childNode,
-        visitedMap,
-        extractedYarnLockV1Pkgs,
-        strictOutOfSync,
-        includeOptionalDeps,
-      );
+    if (localVisited.has(childNode.id)) {
+      if (pruneWithinTopLevel) {
+        const prunedId = `${childNode.id}:pruned`;
+        depGraphBuilder.addPkgNode(
+          { name: childNode.name, version: childNode.version },
+          prunedId,
+          {
+            labels: {
+              scope: node.isDev ? 'dev' : 'prod',
+              pruned: 'true',
+              ...(node.missingLockFileEntry && {
+                missingLockFileEntry: 'true',
+              }),
+            },
+          },
+        );
+        depGraphBuilder.connectDep(node.id, prunedId);
+      } else {
+        depGraphBuilder.connectDep(node.id, childNode.id);
+      }
+      continue;
     }
 
+    depGraphBuilder.addPkgNode(
+      { name: childNode.name, version: childNode.version },
+      childNode.id,
+      {
+        labels: {
+          scope: node.isDev ? 'dev' : 'prod',
+          ...(node.missingLockFileEntry && {
+            missingLockFileEntry: 'true',
+          }),
```

Comment:
Same with this. Though very nitpick, so up to you.

Reply:
I don't know an alternative to this being here it needs to be called every time the code gets to this point. If you are talking repeat adding, adding a repeat node is as close to a noop as we get in js land